lightning/src/routing/router.rs

```diff
@@ -150,11 +160,124 @@ impl cmp::PartialOrd for RouteGraphNode {
 struct DummyDirectionalChannelInfo {
 	cltv_expiry_delta: u32,
 	htlc_minimum_msat: u64,
+	htlc_maximum_msat: Option<u64>,
 	fees: RoutingFees,
 }
 
+/// It's useful to keep track of the hops associated with the fees required to use them,
+/// so that we can choose cheaper paths (as per Dijkstra's algorithm).
+/// Fee values should be updated only in the context of the whole path, see update_value_and_recompute_fees.
+/// These fee values are useful to choose hops as we traverse the graph "payee-to-payer".
+#[derive(Clone)]
+struct PaymentHop {
+	route_hop: RouteHop,
+	/// Liquidity available considering the following limitations:
+	/// - UTXO capacity
+	/// - htlc_maximum_msat (direction-dependent)
+	/// - use of the same channel by other paths per MPP (including fees paid there)
+	/// It does not take into account fees to be paid on the current hop, so
+	/// this amount *should* cover transferring a value AND paying fees (on this channel).
+	available_liquidity_msat: u64,
+	/// Minimal fees required to route to the previous hop via any of its inbound channels.
+	src_lowest_inbound_fees: RoutingFees,
+	/// Fees of the channel used in this hop.
+	channel_fees: RoutingFees,
+	/// All the fees paid *after* this channel on the way to the destination
+	min_total_fee_to_payee: u64,
+	/// Fee paid for the use of the current channel (see channel_fees).
+	/// The value will be actually paid on the previous hop.
+	hop_use_fee_msat: u64,
+	/// Fee required to reach the source node of the current channel (estimate, see src_lowest_inbound_fees)
+	prev_hop_use_estimate_fee_msat: u64,
```

Comment:
agree with `min_total_fee_to_payee` -> `next_hops_fee_msat`.

Don't agree with `hop_use_fee_msat`->`curr_hop_fees_msat`, because I want to highlight it's the fee-paying for the *use* of this channel, but not the fee *charged* on this channel. Also, it's a field of the hop, sure it's "curr".
Same problem with `curr_fee_schema`.

Not sure about `prev_hop_use_estimate_fee_msat` -> `prev_hops_est_fees_msat`. It's talking about just one immediate hop, why "hops"? I'll improve the comment though. Same with `min_fee_schema_prev_hops`.